akka-docs/rst/scala/code/docs/http/scaladsl/SprayJsonExampleSpec.scala

```diff
@@ -47,4 +49,57 @@ class SprayJsonExampleSpec extends WordSpec with Matchers {
       //#
     }
   }
+
+  "second-spray-json-example" in compileOnlySpec {
+    //#second-spray-json-example
+    import akka.actor.ActorSystem
+    import akka.stream.ActorMaterializer
+    import akka.Done
+    import akka.http.scaladsl.server.Route
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+    import spray.json.DefaultJsonProtocol._
+
+    object WebServer {
+
+      // domain model
+      final case class Item(name: String, id: Long)
+      final case class Order(items: List[Item])
+
+      // formats for unmarshalling and marshalling
+      implicit val itemFormat = jsonFormat2(Item)
+      implicit val orderFormat = jsonFormat1(Order)
+
+      // (fake) async database query api
+      def fetchItem(itemId: Long): Future[Option[Item]] = ???
+      def saveOrder(order: Order): Future[Done] = ???
+
+      def main(args: Array[String]) {
+
+        // needed to run the route
+        implicit val system = ActorSystem()
+        implicit val materializer = ActorMaterializer()
+
+        val route: Route =
+          get {
+            pathPrefix("item" / LongNumber) { id =>
+              val maybeItem: Future[Option[Item]] = fetchItem(id)
+              complete(maybeItem)
```

Comment:
Yeah, actually expected that to happen, thought you added such a marshaller that does that the other week?